### PR TITLE
Add keyboard Signum 3.0 for the elite-c

### PR DESCRIPTION
**Pull request by jceb**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by jceb** [comment]
This pr is based on #5430.
- Bootloader is set
- Folder structure is used extensively, it leaves room to add the Signum 3.1 into

I'm not sure about the folder structure. Maybe someone experience can take a look at it. Currently, it only supports the Signum 3.0 with Elite-C micro controller. I thought about adding support for the Teensy and Pro Micro. I'd like to reuse the keymap while changing `config.h` parameters.

**Comment by noroadsleft** [comment]
> `info.json` added - is there a way to preview the output?

1. Go to the [QMK Configurator](https://config.qmk.fm/)
2. Hit `Ctrl`+`Shift`+`I`, in that order (Ctrl before Shift)
3. Upload your saved `info.json` file.

**Comment by jceb** [comment]
Thanks, it looks good!

File: keyboards/signum/3_0/elitec/config.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#define MATRIX_ROW_PINS { D2, D1, F5, B5 }
+#define MATRIX_COL_PINS { B4, D7, D0, E6, D4, F6, F4, F7, B1, B3, C6, B2 }
+#define UNUSED_PINS
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
```
Comment (yanfali) [inline keyboards/signum/3_0/elitec/config.h]: Remove this. It's disabled in rules mk

File: keyboards/signum/3_0/elitec/config.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
```
Comment (yanfali) [inline keyboards/signum/3_0/elitec/config.h]: Ditto. Disabled in rules.mk

Comment (jceb) [reply keyboards/signum/3_0/elitec/config.h]: You're right, I'll remove the two occurrences.

**Comment by jceb** [comment]
@yanfali are you happy with the changes?

File: keyboards/signum/3_0/elitec/rules.mk [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+# Bootloader
+#     This definition is optional, and if your keyboard supports multiple bootloaders of
+#     different sizes, comment this out, and the correct address will be loaded
+#     automatically (+60). See bootloader.mk for all options.
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = yes	# Virtual DIP switch configuration(+1000)
```
Comment (yanfali) [inline keyboards/signum/3_0/elitec/rules.mk]: if you don't need full bootmagic, we've been recommending bootmagic lite.

**Comment by yanfali** [comment]
Can you improve the instructions for running the generate script. Looks like there's requirements that aren't in a default python 3 install, when I tried to run it I saw this error

```➜  default git:(signum3_0) ✗ python3 ./generate_km.py
Traceback (most recent call last):
  File "./generate_km.py", line 4, in <module>
    from sortedcontainers import SortedDict
ModuleNotFoundError: No module named 'sortedcontainers'```

File: keyboards/signum/3_0/elitec/rules.mk (hunk 3 above)
Comment (jceb) [reply keyboards/signum/3_0/elitec/rules.mk]: fix

**Comment by jceb** [comment]
> Can you improve the instructions for running the generate script. Looks like there's requirements that aren't in a default python 3 install, when I tried to run it I saw this error
> 
> ```shell
> ➜  default git:(signum3_0) ✗ python3 ./generate_km.py
> Traceback (most recent call last):
>   File "./generate_km.py", line 4, in <module>
>     from sortedcontainers import SortedDict
> ModuleNotFoundError: No module named 'sortedcontainers'
> ```

done.

**Comment by yanfali** [comment]
Awesome work. Thanks @jceb

File: keyboards/signum/3_0/elitec/readme.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Signum 3.0
+
+![Signum 3.0 Top](https://i.redd.it/vl54yl36bpz11.jpg)
+
+An ergonomic 40% PCB.
+
+Keyboard Maintainer: [Troy Fletcher](http://troyfletcher.net/)
```
Comment (drashna) [inline keyboards/signum/3_0/elitec/readme.md]:
```suggestion
Keyboard Maintainer: [Troy Fletcher](http://troyfletcher.net/)  
```

File: keyboards/signum/3_0/elitec/readme.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+Hardware available here: [Troy Fletcher](http://troyfletcher.net/keyboard_sales.html)
```
Comment (drashna) [inline keyboards/signum/3_0/elitec/readme.md]:
```suggestion
Hardware available here: [Troy Fletcher](http://troyfletcher.net/keyboard_sales.html)  
```

**Comment by drashna** [comment]
Awesome, thanks!

**Pull request merged**